workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,233 @@
+  DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+  PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+  ARTIFACT_NAME: 'Published Site' # TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+  ARTIFACT_ZIP: 'artifacts.zip'
+  DEPLOY_ZIP: 'published_site.zip'
+  IIS_SERVER_CERT_PATH: '' # TODO:  The path to the cert for the IIS server remote management
+  WEBSITE_NAME: '' # TODO:  The unique name of the site particularly for IIS server display
+  WEBSITE_HOST_HEADER: '' # TODO:  The IIS server host header that the site responds to
+  WEBSITE_CERT_PATH: '' # TODO:  The path to the private key cert for the IIS server site binding
+  WEBSITE_CERT_FRIENDLY_NAME: '' # TODO:  The friendly name encoded in the IIS private key cert
+  DEPLOYMENT_FOLDER: '' # TODO:  The local target directory for the web site files, i.e., 'c:\inetpub\website'
+
+jobs:
+  validate-input:
+    runs-on: [ubuntu-20.04]
+    steps:
+      # TODO:  Update if you don't use all these environments in the Development Subscription
+      - name: Clean Environment Input
+        id: clean-env
+        uses: im-open/map-input-action@v1.0.1
+        with:
+          input: ${{ github.event.inputs.environment }}
+          input_map: '{ "Dev": ["dev", "d", "development"], "QA": ["qa", "q"], "Stage": ["stg", "s", "stage"]  }'
+          error_on_no_match: true
+          custom_error_message: 'The environment must be Dev, QA or Stage'
+
+  tech-approval:
+    if: ${{ github.event.inputs.environment }} != "DEV" && ${{ github.event.inputs.environment }} != "QA"
+    runs-on: [ubuntu-20.04]
+    environment: tech-approve
+
+  qa-approval:
+    if: ${{ github.event.inputs.environment }} != "DEV" && ${{ github.event.inputs.environment }} != "QA"
+    needs: tech-approval
+    runs-on: [ubuntu-20.04]
+    environment: qa-approve
+
+  stakeholder-approval:
+    if: ${{ github.event.inputs.environment }} == "PROD"
+    needs: [tech-approval, qa-approval]
+    runs-on: [ubuntu-20.04]
+    environment: stakeholder-approve
+
+  build-deploy:
+    needs: [tech-approval, qa-approval, stakeholder-approval]
+    runs-on: [self-hosted, windows-2019]
+    environment: ${{ github.event.inputs.environment }}
+    steps:
+      - name: Open a PagerDuty Maintenance Window
+        id: open-window
+        uses: im-open/open-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{secrets.PAGERDUTY_API_KEY}}
+          description: '${{ env.PAGERDUTY_WINDOW_DESC }}'
+          minutes: ${{ env.PAGERDUTY_WINDOW_IN_MIN }}
+          service-ids: ${{ env.PAGERDUTY_SERVICE_IDS }}
+
+      - run: |
+          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
+          deploy-the-code.sh
```

I assume that was just copied and pasted form an example?
```suggestion
          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
```